src/main/java/org/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -359,93 +335,91 @@ private void exportEntries() {
     }
 
     private void connectAutomatically() {
-        BackgroundTask
-                .wrap(() -> {
-                    DetectOpenOfficeInstallation officeInstallation = new DetectOpenOfficeInstallation(diag, preferences, dialogService);
+        DetectOpenOfficeInstallation officeInstallation = new DetectOpenOfficeInstallation(ooPrefs, dialogService);
 
-                    Boolean installed = officeInstallation.isInstalled().get();
-                    if (installed == null || !installed) {
-                        throw new IllegalStateException("OpenOffice Installation could not be detected.");
-                    }
-                    return null; // can not use BackgroundTask.wrap(Runnable) because Runnable.run() can't throw exceptions
-                })
-                .onSuccess(x -> connect())
-                .onFailure(ex ->
-                        dialogService.showErrorDialogAndWait(Localization.lang("Autodetection failed"), Localization.lang("Autodetection failed"), ex))
-                .executeWith(Globals.TASK_EXECUTOR);
-    }
+        if (officeInstallation.isExecutablePathDefined()) {
+            connect();
+        } else {
 
-    private void connectManually() {
-        showManualConnectionDialog();
-        if (!dialogOkPressed) {
-            return;
+            FXDialog progressDialog = officeInstallation.initProgressDialog();
```

Comment:
no need to handle the progress dialog on your own, just reuse the methods in `dialogService` (see my recent PR).